src/types.ts

```diff
@@ -1364,6 +1364,33 @@ export enum NFTOwnershipStatus {
     "ARCHIVED" = "ARCHIVED",
 }
 
+export interface ContractUploadRequest {
+    name: string;
+    description: string;
+    longDescription: string;
+    bytecode: string;
+    sourcecode: string;
+    compilerOutputMetadata?: object;
+    docs?: object;
+    abi?: object[];
```

Comment:
Is that really optional? No ABI means no constructor and inability to deploy (in case we need a constructor)